src/NServiceBus.AzureStoragePersistence/SagaPersisters/Azure/AzureSagaPersister.cs

```diff
@@ -206,12 +115,12 @@ void Persist(IContainSagaData saga)
             }
 
             var partitionKey = saga.Id.ToString();
-                
+
             var batch = new TableBatchOperation();
 
             AddObjectToBatch(batch, saga, partitionKey);
 
-            table.ExecuteBatch(batch);
+            await table.ExecuteBatchAsync(batch).ConfigureAwait(true);
```

Comment:
`ConfigureAwait(true)` should be `ConfigureAwait(false)` in all situations according to @timbussmann